addons/udes_stock/models/stock_move_line.py

```diff
@@ -1136,3 +1136,62 @@ def _check_resultant_package_level(self):
                 )
             ):
                 raise ValidationError(_("Maximum package depth exceeded."))
+
+    def _split_and_group_mls_by_quantity(self, maximum_qty):
+        """Split move lines into groups of up to a maximum quantity"""
+        grouped_mls = []
+
+        # Group all (partially) done and cancelled moves first
+        excluded_move_lines = self.filtered(lambda m: m.qty_done or m.state == "done")
+        if excluded_move_lines:
+            grouped_mls.append(excluded_move_lines)
+            self -= excluded_move_lines
+
+        # Sse if any mls are equal to the maximum and add them as individual groups
+        exact_mls = self.filtered(lambda l: l.product_uom_qty == maximum_qty)
+        self -= exact_mls
+        for ml in exact_mls:
+            grouped_mls.append(ml)
+
+        # Next try splitting and grouping to maintain a single ml per group:
+        for ml in self:
+            quantity = ml.product_uom_qty
+            if quantity > maximum_qty:
+                num_full_move_lines = int(quantity // maximum_qty)
+                remainder = quantity % maximum_qty
+                for i in range(num_full_move_lines):
+                    new_ml = ml._split_by_qty(maximum_qty)
+                    grouped_mls.append(new_ml)
+                # If there is not a remainder, ml has already been grouped
+                # so remove from self
+                if not remainder:
+                    self -= ml
+
+        # Finally split and group the remainder to meet the maximum
+        quantity = 0
+        movelines = self.browse()
```

Comment:
Did anyone ever decide on a convention for this? MoveLine=self? If we did not, ignore

Reply:
I think the poll ended up split with no clear decision. So I've assumed either is fine. I personally don't like `MoveLine=self` as that may not be clear whether `MoveLine` is empty or not